include/rtc.h

```diff
@@ -0,0 +1,97 @@
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef RTC_H_
+#define RTC_H_
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdint.h>
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/**
+ * @struct rtc_dev
+ * @brief Structure holding RTC descriptor.
+ */
+struct rtc_dev {
+	/** ID of the real time clock core. */
+	uint8_t id;
+	/** Device specific RTC parameters. */
+	void *extra;
+};
+
+/**
+ * @struct rtc_init_param
+ * @brief Structure holding RTC initialization parameters.
+ */
+struct rtc_init_param {
+	/** ID of the real time clock core. */
+	uint8_t id;
+	/** Device specific RTC parameters. */
+	void *extra;
+};
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+
+/** Initialize the RTC peripheral. */
+int32_t rtc_init(struct rtc_dev **device, struct rtc_init_param *init_param);
+
+/** Free the resources allocated by rtc_init(). */
+int32_t rtc_remove(struct rtc_dev *dev);
+
+/** Start the real time clock. */
+int32_t rtc_start(struct rtc_dev *dev);
```

Comment:
I don't think if someone will need to start and stop an rtc, they can be put in the init and remove. For start and stop functionalities I think a timer should be used. Inspired from https://os.mbed.com/docs/mbed-os/v5.15/mbed-os-api-doxy/mbed__rtc__time_8h_source.html

Reply:
I would argue that it is better to be able to stop and start the RTC in case we need to adjust it's mode of operation, or maybe re-calibrate it.